tools/runtests.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+
+# This script finds and runs Lua unit tests, collects coverage and runs static code analysis.
+
+readonly script_dir=$(dirname "$(readlink -f "$0")")
+if [[ -v $1 ]]
+then
+    readonly base_dir="$1"
+else
+    readonly base_dir=$(readlink -f "$script_dir/..")
+fi
+
+readonly exit_ok=0
+readonly exit_software=2
+readonly src_module_path="$base_dir/src"
+readonly test_module_path="$base_dir/test"
+readonly target_dir="$base_dir/target"
+readonly reports_dir="$target_dir/luaunit-reports"
+readonly luacov_dir="$target_dir/luacov-reports"
+
+function create_target_directories {
+    mkdir -p "$reports_dir"
+    mkdir -p "$luacov_dir"
+}
+
+##
+# Run the unit tests and collect code coverage.
+#
+# Return error status in case there were failures.
+#
+function run_tests {
+    cd "$test_module_path" || exit
```

Comment:
In my opinion a failing `cd` is an error condition. Print a message and `exit 1`

Reply:
Added non-zero exit code. `cd` already prints a error.